src/components/icons/AnimatedIcon.tsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react'
+import { css, cx } from '../../../styled-system/css'
+import { icon } from '../../../styled-system/recipes'
+import { token } from '../../../styled-system/tokens'
+import { AnimatedIconType } from '../../@types/Icon'
+import { ICON_SCALING_FACTOR } from '../../constants'
+import LottieAnimation from './LottieAnimation'
+
+/** Animated Icon with Conditional Lottie Animation. */
+const AnimatedIcon = ({
+  fill,
+  size = 18,
+  style = {},
+  cssRaw,
+  animated,
+  animationData,
+  children,
+  animationComplete,
+}: AnimatedIconType) => {
+  const newSize = size * ICON_SCALING_FACTOR
+  const color = style.fill || fill || token('colors.fg')
+
+  // Local state to manage animation status
+  const [isAnimated, setIsAnimated] = useState(animated)
```

Comment:
Maybe you're setting up for future functionality, but is there a reason you are creating local state instead of just using the `animated` prop?

Reply:
You're right that we could directly use the animated prop for the current functionality. However, the reason I’ve set up local state (isAnimated) is to allow for more flexible control over the animation in future updates.

Reply:
That makes sense. If more control over the animation becomes necessary in future updates, we can always revisit it and reintroduce the state as needed.